lib/al/Library/Layout/LayoutActorUtil.h

```diff
@@ -1,19 +1,116 @@
 #pragma once
 
+#include <math/seadMatrix.h>
 #include <math/seadVector.h>
 
-namespace al {
-class LayoutActor;
-class IUseLayout;
+namespace nn::ui2d {
+class TextureInfo;
+}
 
-bool isActive(al::LayoutActor* actor);
-bool isDead(al::LayoutActor* actor);
+namespace sead {
+class Color4u8;
+}
 
-void killLayoutIfActive(al::LayoutActor* actor);
-void appearLayoutIfDead(al::LayoutActor* actor);
+namespace agl {
+class TextureData;
+}
 
-void calcTrans(sead::Vector3f* trans, const al::IUseLayout*);
-const sead::Vector3f& getLocalTrans(const al::IUseLayout*);
+namespace al {
+class LayoutActor;
+class LayoutPaneGroup;
+class IUseLayout;
+class MessageHolder;
 
-void updateLayoutPaneRecursive(al::LayoutActor* actor);
+bool killLayoutIfActive(al::LayoutActor*);
+bool appearLayoutIfDead(al::LayoutActor*);
+bool isActive(const al::LayoutActor*);
+bool isDead(const al::LayoutActor*);
+void calcTrans(sead::Vector3f*, const al::IUseLayout*);
+sead::Vector3f& getLocalTrans(const al::IUseLayout*);
+sead::Vector3f* getLocalTransPtr(const al::IUseLayout*);
+void calcScale(sead::Vector3f*, const al::IUseLayout*);
+sead::Vector3f& getLocalScale(const al::IUseLayout*);
+void setLocalTrans(al::IUseLayout*, const sead::Vector3f&);
+void setLocalTrans(al::IUseLayout*, const sead::Vector2f&);
+void setLocalScale(al::IUseLayout*, float);
+void setLocalScale(al::IUseLayout*, const sead::Vector2f&);
+void setLocalAlpha(al::IUseLayout*, float);
+void calcPaneTrans(sead::Vector3f*, const al::IUseLayout*, const char*);
+void calcPaneMtx(sead::Matrix34f*, const al::IUseLayout*, const char*);
+void calcPaneTrans(sead::Vector2f*, const al::IUseLayout*, const char*);
+void calcPaneScale(sead::Vector3f*, const al::IUseLayout*, const char*);
+void calcPaneSize(sead::Vector3f*, const al::IUseLayout*, const char*);
+sead::Matrix34f& getPaneMtx(const al::IUseLayout*, const char*);
+sead::Matrix34f* getPaneMtxRaw(const al::IUseLayout*, const char*);
```

Comment:
How did you decide those return types?

Reply:
The code within both of them are the same and returns a pointer. I have to assume that the one that isn't "Raw" returns a reference. I'm noticing now that the "Raw" one has an xref to `alEffectKeeperInitFunction::setupLayoutToEffectKeeper`, which uses it as `const al::Matrix43<float> *`, so I'm changing both of them to be a `Matrix43` and const.